src/net.cpp

```diff
@@ -997,17 +997,29 @@ std::vector<uint8_t> GenerateRandomGarbage() noexcept
 
 } // namespace
 
-V2Transport::V2Transport(NodeId nodeid, bool initiating, int type_in, int version_in, const CKey& key, Span<const std::byte> ent32, Span<const uint8_t> garbage) noexcept :
+void V2Transport::StartSendingHandshake() noexcept
+{
+    AssertLockHeld(m_send_mutex);
+    Assume(m_send_state == SendState::AWAITING_KEY);
+    // Initialize the send buffer with ellswift pubkey + provided garbage.
+    m_send_buffer.resize(EllSwiftPubKey::size() + m_send_garbage.size());
+    std::copy(std::begin(m_cipher.GetOurPubKey()), std::end(m_cipher.GetOurPubKey()), MakeWritableByteSpan(m_send_buffer).begin());
+    std::copy(m_send_garbage.begin(), m_send_garbage.end(), m_send_buffer.begin() + EllSwiftPubKey::size());
```

Comment:
73ad8bbcd23c6bdf3bde0481ae39633ba5b38c1e: could add a note here that we can't clear `m_send_garbage` yet because we haven't constructed the garbage authentication. (could just move the comment you deleted below)